src/Chapter4.hs

```diff
@@ -673,6 +722,29 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data Tree a = Leaf a
+            | Node (Tree a) a (Tree a) deriving (Show, Eq)
+
+data Tree2  = Leaf2 Int
+                | Node2 Tree2 Int Tree2 deriving (Show, Eq)
+
+instance Functor Tree where
+    fmap :: (a -> b) -> Tree a -> Tree b
+    fmap f (Leaf a) = Leaf $ f a
+    fmap f (Node t1 a t2) = Node (fmap f t1) (f a) (fmap f t2)
+
+-- alg:
+-- 1. Call reverseTree for left-subtree.
+-- 2. Call reverseTree for right-subtree.
+-- 3. Swap left and right subtrees.
+reverseTree :: Tree a -> Tree a
+reverseTree (Leaf a) = Leaf a
+reverseTree (Node (Leaf x) a (Leaf y)) = (Node (Leaf y) a (Leaf x))
```

Comment:
This case is actually redundant, because it's a special case of your next line. So you can safely remove it 🧹